### PR TITLE
Fix the bug of address length comparision in rpc.Send()

### PR DESCRIPTION
**Pull request by joezxy**
Suppose there are 5 replicas, and requires 3 successful response.
In the existing code: if opts.N < len(addrs) , it will always fall into error branch.
Should be modified to if len(addrs) < opts.N

### CODE REVIEW

File: rpc/send.go [hunk 1]
```diff
@@ -96,7 +96,9 @@ func (s SendError) CanRetry() bool { return s.canRetry }
 // number of required replies.
 func Send(opts Options, method string, addrs []net.Addr, getArgs func(addr net.Addr) interface{},
 	getReply func() interface{}, context *Context) ([]interface{}, error) {
-	if opts.N < len(addrs) {
+
+	// Return error if provided addrs is less. modified by joezxy 2015/03/07
```
Comment (petermattis) [inline rpc/send.go]: The "modified by joezxy" portion of the comment isn't necessary. That's what a revision control system is for (i.e. `git blame`).

File: rpc/send.go [hunk 2]
```diff
@@ -96,7 +96,9 @@ func (s SendError) CanRetry() bool { return s.canRetry }
...
+	if len(addrs) < opts.N {
```
Comment (petermattis) [inline rpc/send.go]: Mind adding a small test for this.

**Comment by joezxy** [comment]
@petermattis , thanks for your comments. 
the code is modified, and one small testcase is added in send_test.go.
Please help to recheck. :-)

File: rpc/send_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		util.MakeRawAddr("tcp", "127.0.0.1:8082"),
+	}
+
+	testSendOption := Options{
+		N: 5,
+	}
+
+	ret, _ := Send(testSendOption, "", testAddrs, nil, nil, nil)
+	if ret != nil {
+		t.Fatalf("Shorter addrs should have nil and SendError return")
```
Comment (petermattis) [inline rpc/send_test.go]: Like the error message says, you should also check that the returned error is non-nil and a `SendError`.

File: rpc/send_test.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2014 The Cockroach Authors.
```
Comment (petermattis) [inline rpc/send_test.go]: s/2014/2015/g

File: rpc/send_test.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+	"github.com/cockroachdb/cockroach/util"
+)
+
+func TestInvalidAddrLength(t *testing.T) {
+	testAddrs := []net.Addr{
+		util.MakeRawAddr("tcp", "127.0.0.1:8080"),
+		util.MakeRawAddr("tcp", "127.0.0.1:8081"),
+		util.MakeRawAddr("tcp", "127.0.0.1:8082"),
+	}
```
Comment (petermattis) [inline rpc/send_test.go]: `testAddrs` could actually be empty (or nil). There is no need to give it a non-zero length for this test.

File: rpc/send_test.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+	testSendOption := Options{
+		N: 5,
+	}
+
+	ret, _ := Send(testSendOption, "", testAddrs, nil, nil, nil)
```
Comment (petermattis) [inline rpc/send_test.go]: Given that `testSendOption` is only used once, I'd recommend getting rid of the variable and doing something like: `ret, err := Send(Options{N:5}, "", nil, nil, nil, nil)`. The whole test becomes very short, but that's good.

**Comment by petermattis** [comment]
LGTM. Just some small tweaks to the test and we'll get this merged.

**Comment by joezxy** [comment]
@petermattis , please check the test file again. 
I'm new to Golang, thanks for your guidance :-)

**Pull request merged**